invisible_cities/io/mcinfo_io_test.py

```diff
@@ -271,3 +272,20 @@ def test_copy_mc_generator_info(output_tmpdir, ICDATADIR, in_filename, out_filen
             events_out = h5out.root.MC.generators[:]['evt_number']
 
             np.testing.assert_array_equal(events_in, events_out)
+
+
+def test_read_file_with_no_hits(nohits_sim_file):
+
+    filein = nohits_sim_file
+
+    try:
+        load_mcparticles(filein)
+    except IndexError:
+        raise
+
+def test_access_to_particles_in_sns_response_only_file_raises_IndexError(sns_only_sim_file):
+
+    filein = sns_only_sim_file
+
+    with raises(IndexError):
```

Comment:
`IndexError` seems an odd thing to raise here. Don't we want some kind of meaningful IC exception here?

Reply:
`IndexError` is the error raised in `tbl_functions.py` when I try to access the first element of a zero-entries array:

`h5in.root.MC.particles[0]`

which is the test I do to see if the `particle` table is empty. 
Of course, we could add a new exception such as `NoParticleInfoInFile`, I just thought that a readable message could be of use. Can we add readable messages in our IC exceptions?

Reply:
You can add any Python objects as messages inside any Python exception. So use `NoParticleInfoFile` and put whatever string you like as a message inside it.

Reply:
Great, I've implemented it.